wfe2/wfe.go

```diff
@@ -736,19 +737,15 @@ func (wfe *WebFrontEndImpl) processRevocation(
 			"unable to verify provided certificate, empty issuerCertificates")
 	}
 
-	// Try to validate the signature on the provided cert using each of the
-	// known issuer certificates. This is O(n) but we always expect to have
-	// a small number of configured issuers.
-	var validIssuerSignature bool
-	for _, issuer := range wfe.issuerCertificates {
-		if err := providedCert.CheckSignatureFrom(issuer); err == nil {
-			validIssuerSignature = true
-			break
-		}
+	// Try to validate the signature on the provided cert using its corresponding
+	// issuer certificate.
+	issuerNameID := issuance.GetIssuerNameID(providedCert)
+	issuerCert, ok := wfe.issuerCertificates[issuerNameID]
+	if !ok || issuerCert == nil {
```

Comment:
Maps return the zero value when there's no entry, so you can implify this to

```
issuerCert := wfe.issuerCertificates[issuerNameID]
if issuerCert == nil {
...
```

Reply:
I like the version I have here better, because it makes the semantics clearer: "we want to error out if we failed to look something up, *and* if we did look something up but the thing we got was bad". Other situations want only one or the other of those, so having the ability to differentiate is nice.